Add users service proto

rosstimothy opened the pull request:
Introduces the new teleport.users.v1.UsersService and deprecates associated RPCs from the teleport.AuthService.

espadolini commented on api/proto/teleport/users/v1/users_service.proto:
```diff
@@ -0,0 +1,101 @@
+// UsersService provides methods to manage Teleport users.
+service UsersService {
+  // GetUser retrieves a user by name or looks up the current user if requested.
+  rpc GetUser(GetUserRequest) returns (types.UserV2);
+  // ListUsers gets all existing users.
+  rpc ListUsers(ListUsersRequest) returns (ListUsersResponse);
+  // CreateUser creates a user if one does not already exist.
+  rpc CreateUser(CreateUserRequest) returns (types.UserV2);
+  // UpdateUser updates an existing user.
+  rpc UpdateUser(UpdateUserRequest) returns (types.UserV2);
+  // UpsertUser creates a new user or forcefully updates an existing user.
+  rpc UpsertUser(UpsertUserRequest) returns (types.UserV2);
+  // DeleteUser removes an existing user by name.
+  rpc DeleteUser(DeleteUserRequest) returns (google.protobuf.Empty);
```
I still think we should have envelopes for the return values.

rosstimothy committed:
Add users service proto

Introduces the new teleport.users.v1.UsersService and deprecates
associated RPCs from the teleport.AuthService.
